tools/wptrunner/wptrunner/testloader.py

```diff
@@ -31,29 +32,105 @@ def do_delayed_imports():
     from manifest.download import download_from_github  # type: ignore
 
 
-class TestGroupsFile:
-    """
-    Mapping object representing {group name: [test ids]}
-    """
-
-    def __init__(self, logger, path):
+class TestGroups:
+    def __init__(self, logger, path, subsuites):
         try:
             with open(path) as f:
-                self._data = json.load(f)
+                data = json.load(f)
         except ValueError:
             logger.critical("test groups file %s not valid json" % path)
             raise
 
+        self.tests_by_group = defaultdict(set)
         self.group_by_test = {}
-        for group, test_ids in self._data.items():
+        for group, test_ids in data.items():
             for test_id in test_ids:
-                self.group_by_test[test_id] = group
+                id_parts = group.split(":", 1)
+                if len(id_parts) == 1:
+                    group_name = id_parts[0]
+                    subsuite = None
+                else:
+                    subsuites, group_name = id_parts
+                elif subsuite not in subsuites:
+                    raise ValueError(f"Unknown subsuite {subsuite} in group data {group}")
+                self.group_by_test[(subsuite, test_id)] = group_name
+                self.tests_by_group[group_name].add(test_id)
+
+
+def load_subsuites(logger, base_run_info, path: str | None, include_subsuites: set[str]):
```

Comment:
In chromium the supported python version is 3.8. Can we make a change here to accommodate that?

We need to write it as `path: Optional[str]`, and `include_subsuites: Set[str]`.

Similar comments for line 111 and 112.